sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/SharedGatewayHttpClient.java

```diff
@@ -33,7 +32,7 @@ public static HttpClient getOrCreateInstance(HttpClientConfig httpClientConfig,
             }
 
             counter.incrementAndGet();
-            diagnosticsClientConfig.withGatewayHttpClientConfig(sharedGatewayHttpClient.effectiveHttpClientConfig);
+            diagnosticsClientConfig.withGatewayHttpClientConfig(httpClientConfig.toDiagnosticsString());
```

Comment:
should we use sharedGatewayHttpClient.effectiveHttpClientConfig.toDiagnosticsString()

Reply:
I removed the `effectiveHttpClientConfig` from `SharedGatewayHttpClient` class as it was referencing the same thing. Worth taking another look.